src/extension.ts

```diff
@@ -1,11 +1,18 @@
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as mail from '@sendgrid/mail';
+import * as globals from './globals';
+import generateContext from "./context-generator/context-generator";
 import { commands, window, ExtensionContext, workspace, Uri, TextDocument } from 'vscode';
 import { PreviewPanelScope } from './preview-panel-scope';
-import generateContext from "./context-generator/context-generator";
 import { Subject, race } from "rxjs";
 import { debounceTime, filter, take, repeat } from "rxjs/operators";
 import { partialsRegistered, findAndRegisterPartials, watchForPartials } from './partials';
 import { helpersRegistered, findAndRegisterHelpers, watchForHelpers } from './helpers';
+import { HbsTreeItem, HbsContextTreeDataProvider } from './context-data-tree-provider';
+import { getCompiledHtml } from './merger';
 
+let currentEditor: vscode.TextEditor | undefined = undefined;
```

Comment:
I think this files is starting to get kinda bloated. Can we move all code related to thi new feature into a separate module?

Reply:
I moved the code into the respective files